pkg/api/v20170131/validate.go

```diff
@@ -84,6 +84,7 @@ func (a *Properties) Validate() error {
 		if agentPoolProfile.OSType == Windows {
 			switch a.OrchestratorProfile.OrchestratorType {
 			case Swarm:
+			case Kubernetes:
```

Comment:
/we should do validation that this isn't set in 2016-09-30. Unless it is allowed in the current shipped RP.